### ISSUE TITLE
Open source license violations

### ISSUE THREAD

**Issue by joshuaboniface** [jellyfin/jellyfin#610]
In the interest of transparency, this issue will document any instances of software license violations found in the codebase. This will assist us in tracking and correcting these violations, or replacing the relevant code. This applies to all repositories - let's centralize the issue here.

If you find a license violation in *any* source file, please post it here, noting:

1. The repository in question, e.g. `jellyfin`, `jellyfin-web`, etc.
1. The file/directory it was found in, e.g. `MediaBrowser.Blah/MyDir/MyBadViolatingFile.cs`.
1. The upstream project information, if found, including the upstream author.
1. The violation itself, e.g. "Apache2 licensed file in GPLv2 repo" or "No Copyright attribution".

**Comment by cvium** [comment]
## NLangDetect
https://github.com/jellyfin/jellyfin/tree/master/Emby.Server.Implementations/TextEncoding/NLangDetect

Upstream project is https://github.com/marek-stoj/NLangDetect, which is improperly licensed as MIT. The project which it is a port of https://github.com/shuyo/language-detection is Apache2. Furthermore, there's no license included in jellyfin.

**Comment by Bond-009** [comment]
## OpenSubtitles Handler

1. Repository: `jellyfin`
2. Directory: `OpenSubtitlesHandler`
3. Upstream project: https://sourceforge.net/projects/oshandler/
4. Violation: GPLv3+ code in GPLv2 repo

**Comment by Bond-009** [comment]
## MoreLINQ

1. Repository: `jellyfin`
2. File: `MediaBrowser.Model/Extensions/LinqExtensions.cs`
3. Upstream project: https://github.com/morelinq/MoreLINQ
4. Violation: Apache 2 code in GPLv2 repo

**Comment by cvium** [comment]
## libjass

1. Repository: `jellyfin-web`
2. File: `src/bower_components/libjass/libjass.min.js`
3. Upstream project: https://github.com/Arnavion/libjass
4. Violation: Apache 2 code in GPLv2 repo

**Comment by JustAMan** [comment]
> ## cordova-chromecast

After discussion with original authors it was licensed and put under our org: https://github.com/jellyfin/cordova-plugin-chromecast

**Comment by cvium** [comment]
## ServiceStack

1. Repository: `jellyfin`
2. File: I don't know. There are quite a few. One such file is `Emby.Server.Implementations/Services/ServicePath.cs`
3. Upstream project: https://github.com/ServiceStack/ServiceStack
4. Violation: AGPLv3 code in GPL repo

**Comment by oddstr13** [comment]
RSSDP
---------
1. `jellyfin`
2. `RSSDP/LICENSE`
3. https://github.com/Yortw/RSSDP
4. MIT license not included with code; https://github.com/Yortw/RSSDP/blob/master/LICENSE

**Comment by oddstr13** [comment]
Jellyfin release binaries
------------------------------
1. `jellyfin`
2. https://github.com/jellyfin/jellyfin/releases
3. 
4. Licenses not included in binary distribution - at least jellyfin_10.3.5_windows-x64.zip is lacking a LICENSE file, probably missing in other files too.

**Comment by joshuaboniface** [comment]
Closing this out. We've long-since reached the conclusion that we're in good shape license-wise.

The TL;DR is that the source we were using originally, by virtue of not having any *explicit* GPL versioning, as well as the opinions of the Fedora team that *the version specified in a LICENSE file alone is not enough to determine versioning of the GPL*, along with the fact that *the source we were using was only distributable under the GPLv3 precisely *because* of these licensing issues*, means that for all intents and purposes the Emby 3.5.3 source code was licensed GPLv2 *or later*, with its binaries being licensed GPLv3, and thus we may use this source as well as all of Jellyfin as GPLv3. We are thus in compliance with all the licenses of the upstreams within.

**Comment by aleksasiriski** [comment]
Would you ever consider updating the license to GPLv3 or even AGPLv3?

**Comment by joshuaboniface** [comment]
We *can't* change it. We'd have to get acknowledgement from everyone who's ever contributed, which includes a lot of old Emby contributors, the core Emby team, and everyone who's contributed since the fork. The license is also in a very odd situation with regards to GPL v2 vs v3. To quote (part of) a Reddit post I made several months ago on it...

>So our story is one of the messiest I'm aware of but probably far from the only one. I lead the [Jellyfin project](https://github.com/jellyfin), which is a FLOSS fork of Emby from their 3.x days when they were still open source. The code in the repo on GitHub was listed as GPL. But not a specific version of GPL: sure, their LICENSE file had GPL v2, but not a single file had a header clearly declaring whether this was GPLv2-only, GPLv2+, or GPLv3. And according to the legal opinions of the Fedora team, the LICENSE file alone is not enough to declare the version of the GPL that the actual code is under, despite this being precisely how GitHub handles it.

>Now, on its own, this wouldn't be a big deal. We'd just say it's GPLv2 and move on with our lives. But oh no, because the original authors practiced what we can politely call abysmal license hygiene, of the sort described in this article, the repo actually contained a mismash of GPLv2, GPLv2+, GPLv3, MIT, BSD, and most importantly, Apache 2-licensed dependencies, all baked into the code with very little in the way of sensible history or tracking. What this meant, in a practical sense and after hours of pouring over licenses, including going so far as to reach out to the [Software Freedom Law Center](https://softwarefreedom.org/) to try to figure the mess out (It's worth noting too, that theynever got back to us in any form, a major disappointment from an organization that claims to exist precisely for this sort of situation, but I digress), was that only a GPLv3 binary was valid, since GPLv2 and Apache 2 are incompatible. Fun.

>So, we wrote up a big spiel outlining our justification about how the software repo must be GPLv2+, the binaries must be GPLv3, or Emby was violating probably half a dozen licenses to other software tools. And then we sat on it. Mostly because, early on, we (very justifiably given other things that happened) feared for backlash from upstream, still weren't sure of the exact ramifications ourselves, and just generally didn't want to broadcast how fragile our software's license situation was to the world.

So, the tl;dr is:

* The code itself is "GPL unversioned".
* The binaries we distribute (or which are built from the code) are effectively GPLv3.
* Switching to another license (likely AGPLv3) would not be possible without an extremely large coordination effort involving several players who are hostile to us.

**Comment by aleksasiriski** [comment]
Sad to hear that you ended up in such a mess, but at least the project can continue.
